src/codeu/chat/common/Interests.java

```diff
@@ -0,0 +1,52 @@
+package codeu.chat.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collection;
+
+import codeu.chat.util.Serializer;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+
+public final class Interests {
+
+  public static final Serializer<Interests> SERIALIZER = new Serializer<Interests>() {
+
+    @Override
+    public void write(OutputStream out, Interests value) throws IOException {
+
+      Serializers.collection(Uuid.SERIALIZER).write(out, value.interests);
+      Time.SERIALIZER.write(out, value.lastStatusUpdate);
+      Time.SERIALIZER.write(out, value.creation);
+
+    }
+
+    @Override
+    public Interests read(InputStream in) throws IOException {
+
+      return new Interests(
+          Serializers.collection(Uuid.SERIALIZER).read(in),
+          Uuid.SERIALIZER.read(in),
+          Time.SERIALIZER.read(in),
+          Time.SERIALIZER.read(in)
+      );
+
+    }
+  };
+
+  public final Collection<Uuid> interests;
+  public final Uuid id;
```

Comment:
I don't remember exactly what happened, but I remember when @mlk209 and @lysanderp41 were working on this, this fix was the most reasonable solution to an unexpected problem where you needed to get the userid from the interest object itself.